mono/metadata/reflection.c

```diff
@@ -1164,6 +1164,8 @@ method_encode_code (MonoDynamicImage *assembly, ReflectionMethodBuilder *mb)
 	guint32 header_size = 12;
 	MonoArray *code;
 
+	mono_error_init (error);
+
 	if ((mb->attrs & (METHOD_ATTRIBUTE_PINVOKE_IMPL | METHOD_ATTRIBUTE_ABSTRACT)) ||
 			(mb->iattrs & (METHOD_IMPL_ATTRIBUTE_INTERNAL_CALL | METHOD_IMPL_ATTRIBUTE_RUNTIME)))
 		return 0;
```

Comment:
Zero is valid return value.

We should document that the caller must not check the return value for errors.